nx/include/switch/services/mii.h

```diff
@@ -0,0 +1,176 @@
+    MiiSpecialKeyCode_Special  = 0xA523B78F,  ///< Special miis
+} MiiSpecialKeyCode;
+
+typedef struct {
+    Service s;
+} MiiDatabase;
+
+// Mii create ID.
+typedef struct {
+    Uuid uuid;
+} MiiCreateId;
+
+// Mii data structure.
+typedef struct {
+    MiiCreateId create_id;
+    u16 mii_name[10+1]; ///< utf-16be, null-terminated
+    u8 unk_x26;
+    u8 mii_color;
+    u8 mii_sex;
+    u8 mii_height;
+    u8 mii_width;
+    u8 unk_x2b[2];
+    u8 mii_face_shape;
+    u8 mii_face_color;
+    u8 mii_wrinkles_style;
+    u8 mii_makeup_style;
+    u8 mii_hair_style;
+    u8 mii_hair_color;
+    u8 mii_has_hair_flipped;
+    u8 mii_eye_style;
+    u8 mii_eye_color;
+    u8 mii_eye_size;
+    u8 mii_eye_thickness;
+    u8 mii_eye_angle;
+    u8 mii_eye_pos_x;
+    u8 mii_eye_pos_y;
+    u8 mii_eyebrow_style;
+    u8 mii_eyebrow_color;
+    u8 mii_eyebrow_size;
+    u8 mii_eyebrow_thickness;
+    u8 mii_eyebrow_angle;
+    u8 mii_eyebrow_pos_x;
+    u8 mii_eyebrow_pos_y;
+    u8 mii_nose_style;
+    u8 mii_nose_size;
+    u8 mii_nose_pos;
+    u8 mii_mouth_style;
+    u8 mii_mouth_color;
+    u8 mii_mouth_size;
+    u8 mii_mouth_thickness;
+    u8 mii_mouth_pos;
+    u8 mii_facial_hair_color;
+    u8 mii_beard_style;
+    u8 mii_mustache_style;
+    u8 mii_mustache_size;
+    u8 mii_mustache_pos;
+    u8 mii_glasses_style;
+    u8 mii_glasses_color;
+    u8 mii_glasses_size;
+    u8 mii_glasses_pos;
+    u8 mii_has_mole;
+    u8 mii_mole_size;
+    u8 mii_mole_pos_x;
+    u8 mii_mole_pos_y;
+    u8 unk_x57;
+} MiiCharInfo;
+
+/// Initialize mii.
+Result miiInitialize(MiiServiceType service_type);
+
+/// Exit mii.
+void miiExit(void);
+
+/// Gets the Service object for the actual mii service session.
+Service* miiGetServiceSession(void);
+
+/**
+ * @brief Opens a mii database.
+ * @param[in] key_code Mii key code filter.
+ * @param[out] out Database.
+ */
+Result miiOpenDatabase(MiiDatabase *out, MiiSpecialKeyCode key_code);
+
+/**
+ * @brief Returns whether the mii database is updated.
+ * @param[in] db Database.
+ * @param[in] flag Source flag.
+ * @param[out] out_updated Whether the mii database is updated.
+ */
+Result miiDatabaseIsUpdated(MiiDatabase *db, bool *out_updated, MiiSourceFlag flag);
+
+/**
+ * @brief Returns whether the mii database is full.
+ * @param[in] db Database.
+ * @param[out] out_full Whether the mii database is full.
+ */
+Result miiDatabaseIsFull(MiiDatabase *db, bool *out_full);
+
+/**
+ * @brief Returns number of miis in the database with the specified source flag.
+ * @param[in] db Database.
+ * @param[in] flag Source flag.
+ * @param[out] out_count Out mii count.
+ */
+Result miiDatabaseGetCount(MiiDatabase *db, s32 *out_count, MiiSourceFlag flag);
+
+/**
+ * @brief Reads mii charinfo data from the specified source flag.
+ * @param[in] db Database.
+ * @param[in] flag Source flag.
+ * @param[out] out_infos Output mii charinfo array.
+ * @param[in] out_infos_count Amount of mii chainfos to read.
```

Comment:
Param-names in docs were not updated to match the actual param-names.

Reply:
Done